### PR TITLE
Use markdown anchors in the markdown files and update some information there

### PR DESCRIPTION
**Pull request by ericonr**
- Had to remove html tags from those files and use markdown anchors instead
- Changed how files in this repo are linked to each other, GH supports using `[contributing](./CONTRIBUTING.md)`.
- Added some information regarding PRs, which I took from #21481 and one of my own PRs as well. This commit can be split from the rest, but it was easier to include it, since I was already messing with the formatting.

I've tested the hyperlinks in the GH interface, and they should be good to go. The main commit for comments should be f272b49.

### CODE REVIEW

**Comment by Duncaen** [comment]
Not a fan of vmdfmt, this makes the markdown less readable for no real gain, and makes git blame to look at history a lot harder.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -59,8 +59,11 @@ one of the following forms:
    ([example](https://github.com/void-linux/void-packages/commit/176d9655429188aac10cd229827f99b72982ab10)).
 - for package updates, use `<pkgname>: update to <version>.`
    ([example](https://github.com/void-linux/void-packages/commit/b6b82dcbd4aeea5fc37a32e4b6a8dd8bd980d5a3)).
-- for template modifications without a version change, use `<pkgname>: <reason>`
+- for template modifications without a version change, including removals, use
+   `<pkgname>: <reason>`
```
Comment (Chocimier) [inline CONTRIBUTING.md:60]: removes should be rather `<pkgname>: remove package`, reason possibly in body.

**Comment by ericonr** [comment]
@ahesford mentioned they thought it'd be better if the lines wrapped. I can revert it, though.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ericonr) [reply CONTRIBUTING.md:60]: I will add an example for it, then!

**Comment by ericonr** [comment]
@Duncaen and @Chocimier , I updated it. Now it's only some formatting stuff (using anchors instead of html tags, and local files instead of GH links, when possible) plus changes to CONTRIBUTING.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -43,6 +42,10 @@ Once you have built your template file or files, the commit message should have
 
 * for template modifications without a version change, use ```<pkgname>: <reason>``` ([example](https://github.com/void-linux/void-packages/commit/8b68d6bf1eb997cd5e7c095acd040e2c5379c91d)).
 
+* for package removals, use ```<pkgname>: remove package``` ([example](https://github.com/void-linux/void-packages/commit/cbf7353bad4c9297af8b1b395b4d87504e812358)).
```
Comment (Chocimier) [inline CONTRIBUTING.md:45]: This is better example that does not turn package into empty: https://github.com/void-linux/void-packages/commit/83784632d94deee5d038c8e1c4c1dffa922fca21

File: Manual.md [hunk 3]
```diff
@@ -305,12 +296,11 @@ The following functions are defined by `xbps-src` and can be used on any templat
 	`$DESTDIR`. The optional 2nd argument can be used to change the
 	`file name`.
 
-- <a id="vlicense"></a>
  *vlicense()* `vlicense <file> [<name>]`
 
 	Installs `file` into `usr/share/licenses/<pkgname>` in the pkg
 	`$DESTDIR`. The optional 2nd argument can be used to change the
-	`file name`. See [license](#var_license) for when to use it.
+	`file name`. See [license](#mandatory-license) for when to use it.
```
Comment (Chocimier) [inline Manual.md:303]: Formatting broke here, and anchor don't work.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ericonr) [reply CONTRIBUTING.md:45]: Ok, thanks!

File: Manual.md (hunk 3 above)
Comment (ericonr) [reply Manual.md]: Fixed, thanks!

**Comment by ericonr** [comment]
@flexibeast does this cover what you wanted to explain about `[ci skip]`?

**Comment by flexibeast** [comment]
@ericonr: It does, thanks.

**Pull request closed without merging**